synapse/lib/nexus.py

```diff
@@ -463,6 +467,11 @@ async def runMirrorLoop(self, proxy):
                 genr = proxy.getNexusChanges(offs, **opts)
                 async for item in genr:
 
+                    if self.readonly:
+                        logger.error('Unable to consume Nexus events when readonly is set')
```

Comment:
This is going to produce noisy log spam, correct?

Reply:
No, this fini's the client and returns so it shouldn't be spammy, unless we want it to be?